test-specifications/token-service/tokenCreateTransaction.md

```diff
@@ -0,0 +1,1150 @@
+
+| Test no | Name                                                        | Input                                                                                                                                                                    | Expected response                                                                                                                                               | Implemented (Y/N) |
+|---------|-------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates a token with a name that is a valid length          | name="testname", symbol="testsymbol", treasuryAccount=<OPERATOR_ACCOUNT_ID>                                                                                              | The token creation succeeds and the token's name equals "testname".                                                                                             | N                 |
+| 2       | Creates a token with a name that is the minimum length      | name="t", symbol="testsymbol", treasuryAccount=<OPERATOR_ACCOUNT_ID>                                                                                                     | The token creation succeeds and the token's name equals "t".                                                                                                    | N                 |
+| 3       | Creates a token with a name that is empty                   | name="", symbol="testsymbol", treasuryAccount=<OPERATOR_ACCOUNT_ID>                                                                                                      | The token creation fails with a MISSING_TOKEN_NAME response code from the network.                                                                              | N                 |
+| 4       | Creates a token with a name that is the maximum length      | name="This is a really long name but it is still valid because it is 100 characters exactly on the money!!", symbol="testsymbol", treasuryAccount=<OPERATOR_ACCOUNT_ID>  | The token creation succeeds and the token's name equals "This is a really long name but it is still valid because it is 100 characters exactly on the money!!". | N                 |
+| 5       | Creates a token with a name that exceeds the maximum length | name="This is a long name that is not valid because it exceeds 100 characters and it should fail the test!!", symbol="testsymbol", treasuryAccount=<OPERATOR_ACCOUNT_ID> | The token creation fails with a TOKEN_NAME_TOO_LONG response code from the network.                                                                             | N                 |
```

Comment:
Just 26 characters will fail if they are all Linear-B Ideograms, because those characters are 4-bytes each when encoded UTF-8, and characters isn't the limit; the byte length when encoded UTF-8 is the actual limit.